### PR TITLE
Chore: Move Thinkful Advert to TW

### PR DESCRIPTION
**Pull request by KevinMulhern**
Because:
* We are converting all styles on the site to Tailwind.

This commit:
* Refactor advert to use our new card view component
* Replace all styles with Tailwind classes.

Complete the following REQUIRED checkboxes:

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `keyword: brief description of change` format, using one of the following keywords:
  - `Feature` - adds new or amends existing user-facing behaviour
  - `Chore` - changes that have no user-facing value, refactors, dependency bumps, etc
  - `Fix` - bug fixes

Complete the following checkbox ONLY IF it is applicable to your PR. You can complete it later if it is not currently applicable:
-   [x] I have verified all tests and linters pass against my changes, and/or I have included automated tests where applicable

### CODE REVIEW

File: app/views/lessons/_advert.html.erb [hunk 1]
```diff
@@ -1,32 +1,38 @@
...
-      <li class="advert__icons-group-item">
+      <li class="flex items-center flex-col">
         <%= image_tag 'advert/icon-guarantee.svg', alt: 'Job Guarantee', class: 'advert__icon' %>
-        <p class="advert__icon-text">Job Guarantee</p>
+        <p class="font-bold text-gray-800">Job Guarantee</p>
       </li>
 
-      <li class="advert__icons-group-item">
+      <li class="flex items-center flex-col">
         <%= image_tag 'advert/icon-mentor.svg', alt: '1-on-1 Mentorship', class: 'advert__icon' %>
```
Comment (ChargrilledChook) [inline app/views/lessons/_advert.html.erb:33]: We've got some big boys on the image tags
`h-10` on the classes of the 3 timage should give them a matching height to the original 
The alignment is also coming out a bit different but I would almost say it looks better

Do we need to keep the `advert__icon` class for dark mode until everything is completely switched over? Not entirely show how the Tailwind native darkmode works

image

**Review by ChargrilledChook — CHANGES_REQUESTED** [review]

File: app/views/lessons/_advert.html.erb (hunk 1 above)
Comment (KevinMulhern) [reply app/views/lessons/_advert.html.erb]: Oof, I removed the old advert styles last thinking I had moved all the classes over 🙈 

Thanks @ChargrilledChook, nice spot 🔍  updated and ready for re-review.

File: app/assets/stylesheets/dark-mode.css [hunk 2]
```diff
@@ -131,7 +131,7 @@ pre:not(.line-numbers) code {
   color: var(--dark-header-color) !important;
 }
 
-.advert__icon {
+.odin-dark-advert-icon {
```
Comment (ChargrilledChook) [inline app/assets/stylesheets/dark-mode.css:134]: Good idea :ok_hand:

File: app/views/lessons/_advert.html.erb [hunk 3]
```diff
@@ -1,32 +1,38 @@
...
-          <%= image_tag 'advert/thinkful-logo.svg', alt: 'Thinkful Logo', class: 'advert__logo' %>
-          <h2 class="advert__brand-name">Thinkful</h2>
-        </div>
+<%= render CardComponent.new(classes: 'odin-dark-bg-accent') do |card| %>
+  <% card.header do %>
+    <div class="flex justify-between gap-x-16 items-center border-b-2 border-gray-200 pb-6 flex-col space-y-6 md:flex-row md:space-y-0">
+      <h2 class="text-2xl text-gray-500 text-center w-2/3 md:text-left">
+        Are you interested in accelerating your web development learning experience?
+      </h2>
+      <%= link_to "https://www.thinkful.com/?utm_source=odin&utm_medium=cpc&utm_campaign=odin-homepage&utm_content=web-dev-3-icons", class: "button button--primary" do %>
```
Comment (ChargrilledChook) [inline app/views/lessons/_advert.html.erb:7]: Do these button styles need to be changed now, or will that be sorted in a separate story?

**Review by ChargrilledChook — APPROVED** [review]
One question about button styles but otherwise LGTM :rocket:

File: app/views/lessons/_advert.html.erb (hunk 3 above)
Comment (KevinMulhern) [reply app/views/lessons/_advert.html.erb:7]: Sorry mate, I'm probably missing something right in front of me haha. What needs changed about them?

Comment (ChargrilledChook) [reply app/views/lessons/_advert.html.erb:7]: @kevinmulhern No worries mate, I should have been more specific :sweat_smile: 
The BEM classes right at the end, `button button--primary`

Comment (KevinMulhern) [reply app/views/lessons/_advert.html.erb:7]: Ahh I gotcha now, the buttons are already in Tailwind https://github.com/TheOdinProject/theodinproject/blob/main/app/javascript/layouts/buttons.css

**Pull request merged**